group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -705,9 +783,39 @@ public CompletableFuture<OffsetDeleteResponseData> deleteOffsets(
             return FutureUtils.failedFuture(Errors.COORDINATOR_NOT_AVAILABLE.exception());
         }
 
-        return FutureUtils.failedFuture(Errors.UNSUPPORTED_VERSION.exception(
-            "This API is not implemented yet."
-        ));
+        if (!isGroupIdNotEmpty(request.groupId())) {
+            return CompletableFuture.completedFuture(new OffsetDeleteResponseData()
+                .setErrorCode(Errors.INVALID_GROUP_ID.code())
+            );
+        }
+
+        return runtime.scheduleWriteOperation(
+            "delete-offsets",
+            topicPartitionFor(request.groupId()),
+            coordinator -> coordinator.deleteOffsets(context, request)
+        ).exceptionally(exception -> {
+            if (exception instanceof UnknownTopicOrPartitionException ||
+                exception instanceof NotEnoughReplicasException) {
```

Comment:
Error handling codes in lines 555-586 and 797-816 are very similar, an alternative implementation is 

create a util method:

```
private Errors transformExceptionCode(Exception e)
   if (exception instanceof UnknownTopicOrPartitionException ||
                exception instanceof NotEnoughReplicasException) {
                return Errors.COORDINATOR_NOT_AVAILABLE
     }
...
)
deleteGroups : 
exception-> {
return DeleteGroupsRequest.getErrorResultCollection(
                            groupList,
                            transformExceptionCode(e))
}

deleteOffsets: 
exception-> {
return new OffsetDeleteResponseData()
                    .setErrorCode(
                            transformExceptionCode(e).code());
}
```